src/jwt/decode.ts

```diff
@@ -0,0 +1,14 @@
+import { decodeFromBase64Url } from "./utils";
+import type { JWTRegisteredClaims } from "./types";
+
+export function decodeJWT<T extends Record<string, any> = Record<string, any>>(
+  token: string
+) {
+  const parts = token.split(".");
```

Comment:
typescript prevents, js runtime can still fail. This one is little bit more secure:

```suggestion
  const parts = String(token).split(".");
```